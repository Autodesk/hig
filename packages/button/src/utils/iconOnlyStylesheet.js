export default function iconOnlyStylesheet(
  styles,
  props,
  themeData,
  themeMeta
) {
  const iconSize = themeMeta.densityId === "medium-density" ? "18px" : "14px";

  return {
    ...styles,
    button: {
      ...styles.button,
      padding: `${themeData["button.paddingVertical"]}`,
    },
    icon: {
      ...styles.icon,
      left: 0,

      marginRight: 0,
      right: 0,
    },
    iconText: {
      marginLeft: iconSize,
    },
  };
}
