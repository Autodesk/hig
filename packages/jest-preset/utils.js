jest.mock("@hig/utils", () => ({
  generateId: jest.fn(key => `${key}-1`)
}));
