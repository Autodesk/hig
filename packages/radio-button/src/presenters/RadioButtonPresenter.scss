@import '~@hig/styles/mixins/colors';
@import '~@hig/styles/tokens/layers';

.hig__radio-button {
  position: relative;
  display: flex;
  flex-direction: row;
  align-items: center;
  padding: 0 10px;
}

.hig__radio-button__input {
  opacity: 0;
  position: absolute;
  width: 100%;
  height: 100%;
  top: 0;
  left: 0;
  z-index: $checkbox-layer;
}


.hig__radio-button__label {
  font-size: 14px;
  font-weight: normal;
  font-family: 'ArtifaktElement', sans-serif;
  color: color(hig-cool-gray-50);

  margin: 8px;
  line-height: 14px;

  .hig__radio-button--required &::after {
    content: '*';
  }
}

.hig__radio-button__input:checked[disabled], .hig__radio-button__input[disabled]  ~ .hig__radio-button__label {
  color: color(hig-cool-gray-30);
}


.hig__radio-button__wrapper {
  position: relative;
  color: transparent;
  border: 1px solid color(hig-cool-gray-30);
  background-color: transparent;
  text-align: center;
  font-size: 25px;
  line-height: 14px;
  height: 16px;
  width: 16px;
  border-radius: 3px;

  .hig__radio-button__input:hover + & {
    border-color: color(hig-turquoise-50);
  }

  .hig__radio-button__input[disabled] + & {
    border-color: color(hig-cool-gray-30);
    border-style: dotted;
    background-color: transparent;
    pointer-events: none;
  }

  .hig__radio-button__input:focus + & {
    outline: none;
    border-color: color(hig-blue-60);
  }

  .hig__icon{
    position: initial;
  }
}
