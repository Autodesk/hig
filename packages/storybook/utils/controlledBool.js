import { select } from "@storybook/addon-knobs/react";

const options = {
  uncontrolled: "(uncontrolled)",
  true: "Yes",
  false: "No"
};

const valuesByOption = {
  uncontrolled: undefined,
  true: true,
  false: false
};

/**
 * Knob for controlled boolean props
 * @param {string} label
 * @param {boolean|undefined} defaultValue
 * @param {string} group
 * @returns {boolean|undefined}
 */
export default function controlledBool(label, defaultValue, group) {
  const checkedOption = select(label, options, defaultValue, group);

  return valuesByOption[checkedOption];
}
