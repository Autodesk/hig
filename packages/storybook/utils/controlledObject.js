import { text } from "@storybook/addon-knobs/react";

/**
 * Knob for controlled JSON props
 * @param {string} label
 * @param {string|undefined} defaultValue
 * @param {string} group
 * @returns {string|undefined}
 */
export default function controlledObject(label, defaultValue, group) {
  const result = text(label, defaultValue, group);

  return result ? JSON.parse(result) : undefined;
}
