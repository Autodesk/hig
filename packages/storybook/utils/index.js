export { default as controlledBool } from "./controlledBool";
export { default as controlledObject } from "./controlledObject";
export { default as controlledText } from "./controlledText";
export { default as makeSelectOptions } from "./makeSelectOptions";
export { default as translate } from "./translate";
