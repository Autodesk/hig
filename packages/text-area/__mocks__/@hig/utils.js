function generateIdMock(key) {
  return "text-area-1";
}

exports = module.exports = {
  generateId: jest.fn(generateIdMock)
};
