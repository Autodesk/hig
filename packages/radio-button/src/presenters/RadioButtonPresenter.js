import React, { Component } from "react";
import PropTypes from "prop-types";
import { cx, css } from "emotion";
import { ThemeContext } from "@hig/theme-context";
import { createCustomClassNames } from "@hig/utils";

import ButtonPresenter from "./ButtonPresenter";
import stylesheet from "./stylesheet";

export default class RadioButtonPresenter extends Component {
  static propTypes = {
    /**
     * Called when user moves focus from the field
     */
    onBlur: PropTypes.func,
    /**
     * Called when user changes the value of the field
     */
    onChange: PropTypes.func,
    /**
     * Called when user clicks on the field
     */
    onClick: PropTypes.func,
    /**
     * Called when user puts focus on the field
     */
    onFocus: PropTypes.func,
    /**
     * Returns whether or not the button is currently focused
     */
    hasFocus: PropTypes.bool,
    /**
     * Returns whether or not the button is currently hovered
     */
    hasHover: PropTypes.bool,
    /**
     * Returns whether or not the button is currently pressed
     */
    isPressed: PropTypes.bool,
    /**
     * Adds custom/overriding styles
     */
    stylesheet: PropTypes.func
  };

  render() {
    const {
      disabled,
      hasFocus,
      hasHover,
      isPressed,
      stylesheet: customStylesheet,
      ...otherProps
    } = this.props;

    return (
      <ThemeContext.Consumer>
        {({ resolvedRoles }) => {
          const styles = stylesheet(
            {
              isPressed,
              hasFocus,
              hasHover,
              disabled,
              stylesheet: customStylesheet,
              ...this.props
            },
            resolvedRoles
          );

          const { className } = otherProps;
          const inputClassName = createCustomClassNames(className, "input");

          return (
            <div className={cx(css(styles.radioButtonContainer), className)}>
              <input
                {...otherProps}
                disabled={disabled}
                type="radio"
                className={cx(css(styles.radioButtonInput), inputClassName)}
              />
              <ButtonPresenter
                {...otherProps}
                hasFocus={hasFocus}
                hasHover={hasHover}
                isPressed={isPressed}
                disabled={disabled}
                stylesheet={customStylesheet}
              />
            </div>
          );
        }}
      </ThemeContext.Consumer>
    );
  }
}
