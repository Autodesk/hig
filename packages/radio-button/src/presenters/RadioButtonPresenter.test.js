import React from "react";
import renderer from "react-test-renderer";
import { generateId } from "@hig/utils";
import RadioButtonPresenter from "./RadioButtonPresenter";

describe("radio-button/presenters/RadioButtonPresenter", () => {
  afterEach(() => {
    generateId.mockReset();
  });

  [
    {
      description: "renders without props",
      props: {}
    },
    {
      description: "renders with all props",
      props: {
        checked: true,
        disabled: true,
        label: "HELLO",
        name: "RadioButton",
        onBlur: function onBlur() {},
        onClick: function onClick() {},
        onChange: function onChange() {},
        onFocus: function onFocus() {},
        required: "required helper text",
        value: "value"
      }
    },
    {
      description: "renders with className prop",
      props: { title: "Radio Button Class", className: "custom-class" }
    }
  ].forEach(({ description, props: { children, ...otherProps } }) => {
    it(description, () => {
      const presenter = (
        <RadioButtonPresenter {...otherProps}>{children}</RadioButtonPresenter>
      );
      const tree = renderer.create(presenter).toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
});
