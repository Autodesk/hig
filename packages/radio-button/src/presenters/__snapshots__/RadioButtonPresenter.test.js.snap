// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`radio-button/presenters/RadioButtonPresenter renders with all props 1`] = `
.emotion-2 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-0 {
  display: inline-block;
  opacity: 0;
  position: absolute;
  width: 100%;
  height: 100%;
  top: 0;
  left: 0;
  margin: 0;
  padding: 0;
  z-index: 1;
}

.emotion-0:checked + span {
  box-shadow: inset 0 0 0 4px #808080,0 0 0 0px transparent;
}

.emotion-1 {
  opacity: 0.4;
  position: relative;
  color: transparent;
  border: none;
  background-color: transparent;
  text-align: center;
  font-size: 25px;
  line-height: 14px;
  height: 16px;
  width: 16px;
  border-radius: 50%;
  box-shadow: inset 0 0 0 1px  rgba(128,128,128,0.5) ,0 0 0 0px transparent;
  -webkit-transition-duration: 0.3s,0.3s;
  transition-duration: 0.3s,0.3s;
  -webkit-transition-property: box-shadow,background-color;
  transition-property: box-shadow,background-color;
  -webkit-transition-timing-function: cubic-bezier(0.4,0,0.2,1),cubic-bezier(0.4,0,0.2,1);
  transition-timing-function: cubic-bezier(0.4,0,0.2,1),cubic-bezier(0.4,0,0.2,1);
}

<div
  className="emotion-2"
>
  <input
    checked={true}
    className="emotion-0"
    disabled={true}
    label="HELLO"
    name="RadioButton"
    onBlur={[Function]}
    onChange={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    required="required helper text"
    type="radio"
    value="value"
  />
  <span
    className="emotion-1"
  />
</div>
`;

exports[`radio-button/presenters/RadioButtonPresenter renders with className prop 1`] = `
.emotion-2 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-0 {
  display: inline-block;
  opacity: 0;
  position: absolute;
  width: 100%;
  height: 100%;
  top: 0;
  left: 0;
  margin: 0;
  padding: 0;
  z-index: 1;
}

.emotion-0:checked + span {
  box-shadow: inset 0 0 0 4px #808080,0 0 0 0px transparent;
}

.emotion-1 {
  opacity: 1.0;
  position: relative;
  color: transparent;
  border: none;
  background-color: transparent;
  text-align: center;
  font-size: 25px;
  line-height: 14px;
  height: 16px;
  width: 16px;
  border-radius: 50%;
  box-shadow: inset 0 0 0 1px  rgba(128,128,128,0.5) ,0 0 0 0px transparent;
  -webkit-transition-duration: 0.3s,0.3s;
  transition-duration: 0.3s,0.3s;
  -webkit-transition-property: box-shadow,background-color;
  transition-property: box-shadow,background-color;
  -webkit-transition-timing-function: cubic-bezier(0.4,0,0.2,1),cubic-bezier(0.4,0,0.2,1);
  transition-timing-function: cubic-bezier(0.4,0,0.2,1),cubic-bezier(0.4,0,0.2,1);
}

<div
  className="emotion-2 custom-class"
>
  <input
    className="emotion-0 custom-class__input"
    disabled={undefined}
    title="Radio Button Class"
    type="radio"
  />
  <span
    className="emotion-1 custom-class__span"
  />
</div>
`;

exports[`radio-button/presenters/RadioButtonPresenter renders without props 1`] = `
.emotion-2 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-0 {
  display: inline-block;
  opacity: 0;
  position: absolute;
  width: 100%;
  height: 100%;
  top: 0;
  left: 0;
  margin: 0;
  padding: 0;
  z-index: 1;
}

.emotion-0:checked + span {
  box-shadow: inset 0 0 0 4px #808080,0 0 0 0px transparent;
}

.emotion-1 {
  opacity: 1.0;
  position: relative;
  color: transparent;
  border: none;
  background-color: transparent;
  text-align: center;
  font-size: 25px;
  line-height: 14px;
  height: 16px;
  width: 16px;
  border-radius: 50%;
  box-shadow: inset 0 0 0 1px  rgba(128,128,128,0.5) ,0 0 0 0px transparent;
  -webkit-transition-duration: 0.3s,0.3s;
  transition-duration: 0.3s,0.3s;
  -webkit-transition-property: box-shadow,background-color;
  transition-property: box-shadow,background-color;
  -webkit-transition-timing-function: cubic-bezier(0.4,0,0.2,1),cubic-bezier(0.4,0,0.2,1);
  transition-timing-function: cubic-bezier(0.4,0,0.2,1),cubic-bezier(0.4,0,0.2,1);
}

<div
  className="emotion-2"
>
  <input
    className="emotion-0"
    disabled={undefined}
    type="radio"
  />
  <span
    className="emotion-1"
  />
</div>
`;
